### PR TITLE
Add ISMIP6 Antarctic ocean regions

### PR DESCRIPTION
**Pull request by xylar**
These features are the ones used to select CMIP5/6 models for ocean forcing in the ISMIP6 project.  They will be used for analysis in a paper that is in progress.

This merge also adds a creation script for these features.

### CODE REVIEW

**Comment by xylar** [comment]
@alicebarthel, this PR adds the features we discussed to `geometric_features`.  It will require another step (which I will take care of) to be able to use them in MPAS-Analysis.

For now, I would like you to take a careful look at each of these 6 (just on GitHub is fine) and make sure they make sense:
https://github.com/MPAS-Dev/geometric_features/blob/8459c5f498ffaef3e7ca365a7aef407f09caf5a6/geometric_data/ocean/region/ISMIP6_Amery_Sector_Shelf/region.geojson
https://github.com/MPAS-Dev/geometric_features/blob/8459c5f498ffaef3e7ca365a7aef407f09caf5a6/geometric_data/ocean/region/ISMIP6_Amundsen_Sea_Shelf/region.geojson
https://github.com/MPAS-https://github.com/MPAS-Dev/geometric_features/blob/8459c5f498ffaef3e7ca365a7aef407f09caf5a6/geometric_data/ocean/region/ISMIP6_Totten_Region_Shelf/region.geojson
https://github.com/MPAS-Dev/geometric_features/blob/8459c5f498ffaef3e7ca365a7aef407f09caf5a6/geometric_data/ocean/region/ISMIP6_Weddell_Sea_Shelf/region.geojson
Dev/geometric_features/blob/8459c5f498ffaef3e7ca365a7aef407f09caf5a6/geometric_data/ocean/region/ISMIP6_Ross_Sea/region.geojson

The Ross is divided into 2 halves because it crosses the date line (antemeridian).  That's just a weird requirement of geojson files.

The other feedback I need from you is what zmin and zmax should be.  Currently, they're -1000 and -200, but that's just because that happened to be what we used for the shelf in our previous Antarctic shelf regions.

File: feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+    fcContour = fc_world.difference(fcContour)
+
+    fcShelf = fc.difference(fcContour)
+
+    props = fcShelf.features[0]['properties']
+    props['name'] = props['name'] + ' Shelf'
+    props['tags'] = props['tags'] + ';Shelf'
+    props['zmin'] = -1000.
+    props['zmax'] = -200.
```
Comment (xylar) [inline feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py:200]: Hes'e where we need to set the `zmin` and `zmax` we want.  Of course, we can always do something else in MPAS-Analysis but we might as well start with something useful.

Comment (alicebarthel) [reply feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py:200]: Thanks Xylar. The depth range we are looking for is 1500-200, so zmax is ok but zmin should be changed to -1500. Can you confirm the regions were set with a 1500m depth threshold? Both the averaging and the spatial regions have a 1500m limit.

**Comment by alicebarthel** [comment]
Xylar, the regions look good. Maybe some marginal differences but I reckon due to the plotting and topo, as the values shown in the raw geojson files correspond to the ones I have given. Please just clarify the continental shelf threshold when you can (it should be 1500m).

File: feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py (hunk 1 above)
Comment (xylar) [reply feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py:200]: Okay, great, I'll make that change.

File: feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    author = 'Xylar Asay-Davis, Alice Barthel, Nicolas Jourdain'
+    tags = 'Antarctic;ISMIP6'
+
+    # make a geometric features object that knows about the geometric data
+    # cache up a couple of directories
+    gf = GeometricFeatures('../../geometric_data')
+
+    bedmap2_bin_to_netcdf('bedmap2.nc')
+
+    fcContour1500 = get_longest_contour(contourValue=-1500., author=author)
```
Comment (xylar) [inline feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py:215]: @alicebarthel, here's where the depth of the contour that splits each region gets defined.  It's -1500 m, as you specified.

Comment (alicebarthel) [reply feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py:215]: Ok, great. Thanks for clarifying!

File: feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+
+    fcContour = fc_world.difference(fcContour)
+
+    fcShelf = fc.difference(fcContour)
+
+    props = fcShelf.features[0]['properties']
+    props['name'] = props['name'] + ' Shelf'
+    props['tags'] = props['tags'] + ';Shelf'
+    props['zmin'] = -1500.
+    props['zmax'] = -200.
```
Comment (xylar) [inline feature_creation_scripts/ismip6_antarctic_ocean_regions/ismip6_antarctic_ocean_regions.py:200]: @alicebarthel, can you verify that these are right?

**Review by alicebarthel — APPROVED** [review]

File: geometric_features/features_and_tags.json [hunk 4]
```diff
@@ -1237,6 +1237,34 @@
...
+        "Shelf"
+      ],
+      "ISMIP6 Dronning Maud Land Shelf": [
+        "Antarctic",
+        "ISMIP6",
+        "Shelf"
+      ],
+      "ISMIP6 Ross Sea": [
+        ""
+      ],
```
Comment (xylar) [inline geometric_features/features_and_tags.json:1257]: This needs fixing!

**Comment by xylar** [comment]
@alicebarthel, thanks for the review!  I'm going to do a test to make sure this works in MPAS-Analysis before I merge this PR and make the release.  It's a lot of work if I find out later that I made a mistake...

**Comment by xylar** [comment]
Merging based on testing in https://github.com/MPAS-Dev/MPAS-Analysis/pull/777

**Pull request merged**